packages/edit-site/src/components/global-styles/utils.js

```diff
@@ -11,6 +11,30 @@ export function getVariationClassName( variation ) {
 	return `is-style-${ variation }`;
 }
 
+/**
+ * Returns a new index based on the presets and slug prefix.
+ *
+ * @param {Array}  presets    The array of presets.
+ * @param {string} slugPrefix The prefix for the slug.
+ *
+ * @return {number} The new index.
+ */
+export function getNewIndexFromPresets( presets, slugPrefix ) {
+	const nameRegex = new RegExp( `^${ slugPrefix }([\\d]+)$` );
+	return presets.reduce( ( previousValue, currentValue ) => {
+		if ( typeof currentValue?.slug === 'string' ) {
+			const matches = currentValue?.slug.match( nameRegex );
+			if ( matches ) {
+				const id = parseInt( matches[ 1 ], 10 );
+				if ( id >= previousValue ) {
+					return id + 1;
+				}
+			}
+		}
+		return previousValue;
+	}, 1 );
```

Comment:
Minor optional feedback (it doesn't need to be addressed), but I think the function can be more self documenting with some small changes to the variable naming and if the increment happens at the end:
```jsx
	const highestPresetValue = presets.reduce( ( currentHighest, preset ) => {
		if ( typeof preset?.slug === 'string' ) {
			const matches = preset?.slug.match( nameRegex );
			if ( matches ) {
				const id = parseInt( matches[ 1 ], 10 );
				if ( id > currentHighest ) {
					return id;
				}
			}
		}
		return currentHighest;
	}, 0 );
	return highestPresetValue + 1;
```

I haven't tested it though, so not sure if this code still works 😄

Reply:
Thanks for pointing it out. I completely agree with your suggestion and also I've tested these changes and it is working as expected.